core/app/models/spree/product/scopes.rb

```diff
@@ -192,7 +192,7 @@ def self.property_conditions(property)
 
     # Can't use add_search_scope for this as it needs a default argument
     def self.available(available_on = nil, currency = nil)
-      not_discontinued.joins(:master => :prices).where("#{Product.quoted_table_name}.available_on <= ?", available_on || Time.current)
+      not_discontinued.joins(master: :prices).where("#{Product.quoted_table_name}.available_on <= ?", available_on || Time.current).uniq
```

Comment:
@nishant-cyro As this is itself defined in product/scopes. I do not see a reason to add scope for this.